testscript/cmd.go

```diff
@@ -409,6 +411,27 @@ func (ts *TestScript) cmdSymlink(neg bool, args []string) {
 	ts.Check(os.Symlink(args[2], ts.MkAbs(args[0])))
 }
 
+// cmdUNIX2DOS converts files from UNIX line endings to DOS line endings.
+func (ts *TestScript) cmdUNIX2DOS(neg bool, args []string) {
+	if neg {
+		ts.Fatalf("unsupported: ! unix2dos")
+	}
+	if len(args) < 1 {
+		ts.Fatalf("usage: unix2dos paths...")
+	}
+	for _, arg := range args {
+		filename := ts.MkAbs(arg)
+		data, err := ioutil.ReadFile(filename)
+		if err != nil {
+			ts.Fatalf("%s: %v", filename, err)
+		}
+		data = bytes.Join(bytes.Split(data, []byte{'\n'}), []byte{'\r', '\n'})
```

Comment:
I wonder about:
```
data = bytes.ReplaceAll(data, []byte{\n'}, []byte{'\r', '\n'})
```
which is both shorter and more efficient.

But also, I wonder if it would be better if this command was idempotent.

How about:

```
func unix2DOS(data []byte) []byte {
	var out []byte
	for scan := bufio.NewScanner(); scan.Scan(); {
		out = append(out, scan.Bytes())
		out = append(out, '\r', '\n')
	}
	return out
}
```

That way you can call unix2DOS without thinking too much.

Reply:
Thanks, done.